### PR TITLE
Kill the container we just started if we fail to allocate an IP address

### LINKED ISSUE #921: `weave launch-dns|start` w/o running weave reports error but still launches
````
$ weave launch-dns
weave container is not present. Have you launched it?
$ weave launch-dns
weavedns is already running.
````

Ditto for `weave start` and `weave --local run`.

The failure is in attempting to obtain an IP address.

In #831 we fixed a similar issue wrt network configuration by introducing a `with_container_netns*_or_die*`. Perhaps we need a `ipam_cidrs_or_die`.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #921 and #889

### CODE REVIEW

File: weave [hunk 1]
```diff
@@ -788,7 +788,9 @@ ipam_cidrs() {
     IPAM_CIDRs=""
     if [ $# -eq 0 ] ; then
         # If no addresses passed in, get one in the default subnet
-        IPAM_CIDRS=$(call_weave POST /ip/$CONTAINER_ID)
+        if ! IPAM_CIDRS=$(call_weave POST /ip/$CONTAINER_ID) ; then
```
Comment (paulbellamy) [inline weave]: Isn't this the same as just: `IPAM_CIDRS=$(call_weave POST /ip/$CONTAINER_ID)`

**Comment by rade** [comment]
Should the proxy now call `weave attach --or-die` instead of doing its own cleaning up?

File: weave (hunk 1 above)
Comment (bboreham) [inline weave]: No; in the absence of anyone looking at the return value from `call_weave` its return value is overwritten by the next command, i.e. `[`.
This doesn't matter when `set -e` is in effect, but my next change calls `call_cidr` in an if-statement so removing the instant-exit behaviour.

**Comment by paulbellamy** [comment]
+1 for proxy using `weave attach --or-die`

File: weave (hunk 1 above)
Comment (paulbellamy) [inline weave]: Ah, right. Thanks.

**Pull request merged**